lib/transaction.js

```diff
@@ -26,6 +30,19 @@ var Transaction = module.exports = function(sequelize, options) {
   } else {
     this.id = this.name = Utils.generateUUID();
   }
+
+  // `transaction.promise` implementation #2805
+  // main propose is to allow chainable logic inside hooks
+  this.promise = new Promise(function () {});
```

Comment:
The resolver function probably needs to save references to the resolve and reject functions on the transaction, similar to what the sequelizeResolver function in promise.js is currently doing. There is no way to resolve / reject a promise from the "outside"